cfme/cloud/flavor.py

```diff
@@ -108,11 +143,71 @@ class Flavor(BaseEntity, Taggable):
     name = attr.ib()
     provider = attr.ib()
 
+    def delete(self, wait=True, cancel=False):
+        """Delete current falvor """
+        view = navigate_to(self, 'Details')
+        view.toolbar.configuration.item_select('Remove Flavor', handle_alert=not cancel)
+        view.flash.assert_success_message(
+            'Delete of Flavor "{}" was successfully initiated.'.format(self.name))
+
+        if wait:
+            """Wait for flavor to disappear"""
+            wait_for(lambda: not self.exists,
+                     timeout=600,
+                     message='Wait for flavor to disappear',
+                     delay=20,
+                     fail_func=self.refresh)
+
+    def refresh(self):
+        """Refresh provider relationships and browser"""
+        self.provider.refresh_provider_relationships()
+        self.browser.refresh()
+
+    @property
+    def exists(self):
+        try:
+            navigate_to(self, 'Details')
+            return True
+        except FlavorNotFound:
+            return False
+
 
 @attr.s
 class FlavorCollection(BaseCollection):
     ENTITY = Flavor
 
+    def create(self, name, provider, ram, vcpus, disk, swap, rxtx, is_public=True, cancel=False,
+            wait=True):
+        """Create new falvor"""
+        view = navigate_to(self, 'Add')
+        form_params = {'provider': provider.name,
```

Comment:
If all of these parameters are what define a flavor entity that can be created in MIQ, then the entity model should include them as attributes. Otherwise there's no way to model a flavor that we create, and assert any of its attributes against what is created on the provider itself. (verifying MIQ behavior.